src/bin/bench.rs

```diff
@@ -159,9 +165,88 @@ fn main() {
 
 fn run(config: Config) -> Result<(), Box<Error>> {
     println!("running benchmarking suite...");
+
+    // the thread pool we use to perform our operations on the tree
+    let pool = rayon::ThreadPool::new(rayon::Configuration::new().num_threads(config.num_threads))
+        .unwrap();
+
+    // create a default rsdb config
+    let rsdb_config = rsdb::Config::default();
+    let mut tree = rsdb_config.tree();
+
+    pool.install(|| perform_operations(&mut tree, config));
+
     Ok(())
 }
 
+fn perform_operations(tree: &mut Tree, config: Config) {
```

Comment:
does tree need to be `&mut` or should it just be `&`? Rayon will be passing a reference to the tree to multiple threads, so I think it needs to be an immutable reference. The tree handles mutability by using atomic operations, so it is actually made to be shared between multiple threads without a mutex or needing a mutable reference.